packages/scheduler/src/Scheduler.js

```diff
@@ -605,7 +605,22 @@ if (typeof window !== 'undefined' && window._schedMock) {
   window.addEventListener('message', idleTick, false);
 
   var animationTick = function(rafTime) {
-    isAnimationFrameScheduled = false;
+    if (scheduledCallback !== null) {
+      // Eagerly schedule the next animation callback at the beginning of the
+      // frame. If the scheduler queue is not empty at the end of the frame, it
+      // will continue flushing inside that callback. If the queue *is* empty,
+      // then it will exit immediately. Posting the callback at the start of the
+      // frame ensures it's fired within the earliest possible frame. If we
+      // waited until the end of the frame to post the callback, we risk the
+      // browser skipping a frame and not firing the callback until the frame
+      // after that.
+      requestAnimationFrameWithTimeout(animationTick);
+    } else {
+      // No pending work. Exit.
+      isAnimationFrameScheduled = false;
```

Comment:
We can now unify these checks so we can get rid of this flag and just use `scheduledCallback` as the flag.

Reply:
I thought this would work but now I don't think it does. There are three callbacks: the one scheduled with `requestAnimationFrame` (represented by `isAnimationFrameScheduled`), the `message` event handler (represented by `isIdleScheduled`), and the one that actually contains the work (`scheduledCallback`). Need all three concepts. For example, as added by this PR, sometimes an animation frame is scheduled even if there's no actual work scheduled. So I think all three are needed.

I'll rename these a bit so it's less confusing.